xtask/src/vulnerabilities.rs

```diff
@@ -0,0 +1,291 @@
+fn cargo_careful() {
+    if is_current_toolchain_nightly() {
+        ensure_cargo_command_is_installed("cargo-careful");
+        rustup_add_component("rust-src");
+        // prepare careful sysroot
+        group!("Cargo: careful setup");
+        run_cargo(
+            "careful",
+            Params::from(["setup"]),
+            HashMap::new(),
+            "Cargo sysroot should be available",
+        );
+        endgroup!();
+        // Run cargo careful
+        group!("Cargo: run careful checks");
+        run_cargo(
+            "careful",
+            Params::from(["test"]),
+            HashMap::new(),
+            "Cargo careful should be installed and it should correctly run",
+        );
+        endgroup!();
+    } else {
+        error!(
+            "You must use 'cargo +nightly' to run nightly checks.
+Install a nightly toolchain with 'rustup toolchain install nightly'."
+        )
+    }
+}
+
+// Represents the various sanitizer available in nightly compiler
+// source: https://doc.rust-lang.org/beta/unstable-book/compiler-flags/sanitizer.html
+#[allow(clippy::upper_case_acronyms)]
+enum Sanitizer {
+    Address,
+    CFI,
+    HWAddress,
+    KCFI,
+    Leak,
+    Memory,
+    MemTag,
+    SafeStack,
+    ShadowCallStack,
+    Thread,
+}
+
+impl fmt::Display for Sanitizer {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Sanitizer::Address => write!(f, "AddressSanitizer"),
+            Sanitizer::CFI => write!(f, "ControlFlowIntegrity"),
+            Sanitizer::HWAddress => write!(f, "HWAddressSanitizer"),
+            Sanitizer::KCFI => write!(f, "KernelControlFlowIntegrity"),
+            Sanitizer::Leak => write!(f, "LeakSanitizer"),
+            Sanitizer::Memory => write!(f, "MemorySanitizer"),
+            Sanitizer::MemTag => write!(f, "MemTagSanitizer"),
+            Sanitizer::SafeStack => write!(f, "SafeStack"),
+            Sanitizer::ShadowCallStack => write!(f, "ShadowCallStack"),
+            Sanitizer::Thread => write!(f, "ThreadSanitizer"),
+        }
+    }
+}
+
+impl Sanitizer {
+    const DEFAULT_RUSTFLAGS: &'static str = "-Copt-level=3";
+
+    fn run_tests(&self) {
+        if is_current_toolchain_nightly() {
+            group!("Sanitizer: {}", self.to_string());
+            if self.is_target_supported() {
+                let envs = vec![
+                    (
+                        "RUSTFLAGS",
+                        format!("{} {}", self.flags(), Sanitizer::DEFAULT_RUSTFLAGS),
+                    ),
+                    ("RUSTDOCFLAGS", self.flags().to_string()),
+                ];
+
+                let features = self.cargo_features();
+                let mut args = vec!["--", "--color=always", "--no-capture"];
+                args.extend(features);
+
+                run_cargo(
+                    "test",
+                    args.into(),
+                    envs.into_iter().collect(),
+                    "Failed to run cargo fmt",
```

Comment:
Is it `test` or `fmt` ?

Reply:
Good catch. The expect messages are not conventional also. This parameter ends up being an expect message. I'll fix this next time I work in xtasks.